src/metabase/api/advanced_computation.clj

```diff
@@ -55,6 +61,47 @@
   [[first-query & more-queries] context]
   (qp/process-query first-query (append-queries-context context more-queries)))
 
+(defn run-query
+  "Run the pivot query"
+  ([query]
+   (run-query query {:executed-by api/*current-user-id*}))
+
+  ([query info]
+   (qp.streaming/streaming-response [qp-context :api]
+     (qp.store/with-store
+       (let [main-breakout           (:breakout (:query query))
+             col-determination-query (-> query
+                                         ;; TODO: move this to a bitmask or something that scales better / easier to use
+                                         (assoc-in [:query :expressions] {"pivot-grouping" [:ltrim (json/generate-string main-breakout)]})
+                                         (assoc-in [:query :fields] [[:expression "pivot-grouping"]]))
+             all-expected-cols       (qp/query->expected-cols (qp/query->preprocessed col-determination-query))
+             all-queries             (pivot/generate-queries query)
+             first-query             (first all-queries)
```

Comment:
this could simply be 
`[first-query & rest-queries] (pivot/generate-queries query)`